src/databricks/labs/lsql/core.py

```diff
@@ -182,6 +189,7 @@ def __init__(  # pylint: disable=too-many-arguments,too-many-positional-argument
             ColumnInfoTypeName.TIMESTAMP: self._parse_timestamp,
         }
 
+    @retried(is_retryable=_is_retryable_delta_concurrent_append, timeout=timedelta(seconds=10))
```

Comment:
@nfx : Is this what you thinking off?

I have to think about the implications of always retrying this in lsql, maybe we should only retry within UCX isntead

Reply:
I think there needs to be a flag to control this, and should default to off: it's not necessarily safe to blindly retry arbitrary SQL.

In general any time we do a 'read-modify-write' cycle, everything needs to start again from the `read` part because the `modify` (and `write`) often depend on it. Sometimes the `read` and `modify` bits are within the same SQL statement as the `write`, in which case this is safe. But often this is part of application code before we get to SQL and that may need to be restarted. In this situation only the application knows what to do.

Irrespective of this, whatever we do here also needs to end up in the `.save_table()` implementations: these don't all pass through `.execute()` and the same thing can happen there.

Reply:
1) we need to support this in `RuntimeBackend`, separately
2) i think we need to throw a predefined common exception, to @asnare point

Reply:
I will move the retry to the `RuntimeBackend`. What "predefined common exception" would you suggest to throw? I looked at the `sdk.errors` and did not see one that really applied here